### PR TITLE
Allow to generate pkcs12 containing cert and private key

### PR DESCRIPTION
**Pull request by Annih**
Hello,

This is a small PR to allow generation of a p12 file containing both certificate and private key.
Again I tried to add useful tests and a bit of documentation, but let me know if you want me to add more.

@zuazo  this one should already follow rubocop normally :)

cc. @aboten

### CODE REVIEW

File: libraries/resource_ssl_certificate_pkcs12.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        end
+
+        protected
+
+        def default_pkcs12_path
+          lazy { read_namespace(%w(pkcs12_path)) }
+        end
+
+        def default_pkcs12_passphrase
+          lazy { read_namespace(%w(pkcs12_passphrase)) }
```
Comment (zuazo) [inline libraries/resource_ssl_certificate_pkcs12.rb]: We also need to update the [*README* namespace documentation](https://github.com/criteo-forks/ssl_certificate-cookbook/blob/pkcs12_generation_support/README.md#namespaces) with these:

```
| `namespace['pkcs12_path']`                   | Optional PKCS12 full path.
| `namespace['pkcs12_passphrase']`             | Optional PKCS12 passphrase.
```

File: README.md [hunk 2]
```diff
@@ -407,6 +409,8 @@ When a namespace is set in the resource, it will try to read the following attri
 | `namespace['ssl_chain']['content']`                | Intermediate certificate chain content used when reading from attributes.
 | `namespace['ca_cert_path']`                        | Certificate Authority full path.
 | `namespace['ca_key_path']`                         | Key Authority full path.
+| `namespace[pkcs12_path']`                          | Optional PKCS12 full path.
+| `namespace[pkcs12_passphrase']`                    | Optional PKCS12 passphrase.
```
Comment (zuazo) [inline README.md]: Opening single quotes are missing here.

**Comment by Annih** [comment]
Sorry, I was not very focus on my last fix =]

**Comment by zuazo** [comment]
Awesome! Thank you @Annih.

**Comment by zuazo** [comment]
It seems that this implementation does not work properly, I tried to fix it in a634a0c. Let me know if I broke something.

**Comment by Annih** [comment]
Hello @zuazo,

What was the issue  you tried to fix? I didn't get it with your patch, maybe something wrong with the passphrase?
I let you a [comment on your commit](https://github.com/zuazo/ssl_certificate-cookbook/commit/a634a0cf0169c14e1cc8b0ebdd4ccbb5063b3af8#commitcomment-13062684), because I think you might have change this behavior.

I won't be able to test, or update the other PR before 16th of september, sorry for the bug :(

**Comment by zuazo** [comment]
For example, running your kitchen tests gave me the following error:

```
$ git checkout 95fa07c
HEAD is now at 95fa07c... Merge pull request #17 from criteo-forks/pkcs12_generation_support
$ kitchen test default-debian-78
[...]
       Recipe: ssl_certificate_test::default
         * ssl_certificate[ssl-certificate.local] action create
         Recipe: <Dynamically Defined Resource>
           * file[ssl-certificate.local SSL certificate key] action create
             - create new file /etc/ssl/private/ssl-certificate.local.key
             - update content in file /etc/ssl/private/ssl-certificate.local.key from none to 78c502
             - suppressed sensitive resource
             - change mode from '' to '0600'
             - change owner from '' to 'root'
             - change group from '' to 'root'
           * file[ssl-certificate.local SSL public certificate] action create
             - create new file /etc/ssl/certs/ssl-certificate.local.pem
             - update content in file /etc/ssl/certs/ssl-certificate.local.pem from none to da85aa
             - suppressed sensitive resource
             - change mode from '' to '0644'
             - change owner from '' to 'root'
             - change group from '' to 'root'
           * file[ssl-certificate.local SSL intermediary chain combined certificate] action create
             - create new file /etc/ssl/certs/ssl-certificate.local.pem.chained.pem
             - update content in file /etc/ssl/certs/ssl-certificate.local.pem.chained.pem from none to da85aa
             - suppressed sensitive resource
             - change mode from '' to '0644'
             - change owner from '' to 'root'
             - change group from '' to 'root'
         
         * file[ssl-certificate.local SSL certificate key] action nothing (skipped due to action :nothing)
         * file[ssl-certificate.local SSL public certificate] action nothing (skipped due to action :nothing)
         * file[ssl-certificate.local SSL intermediary chain combined certificate] action nothing (skipped due to action :nothing)
[...]
-----> Running bats test suite
 ✓ creates dummy1 certificate
 ✓ creates dummy1 certificate key
 ✓ creates dummy2 certificate
 ✓ creates dummy2 certificate key
 ✓ creates dummy3 certificate
 ✓ creates dummy3 certificate key
 ✓ sets dummy4 certificate country
 ✓ creates dummy4 certificate key
 ✓ creates dummy5 certificate from a data bag
 ✓ creates dummy5 certificate key from a data bag
 ✓ creates dummy6 certificate from node attributes
 ✓ creates dummy6 certificate key from node attributes
 ✓ creates dummy8 certificate
 ✓ creates dummy8 certificate key
 ✗ creates dummy8 PKCS12 file
          (in test file /tmp/verifier/suites/bats/certificates.bats, line 99)
            `openssl pkcs12 -in "${CERT_PATH}/dummy8.p12" -noout -passin pass:' failed
          Error opening input file /etc/ssl/certs/dummy8.p12
          /etc/ssl/certs/dummy8.p12: No such file or directory
 ✓ creates the apache2 virtualhost
 ✓ sets default web ssl protocol
 ✓ sets default web ssl cipher suite
```

As you can see, the PKCS12 was not generated.

Don't worry about it. You have done a great job. Someday I have to refactor the libraries or document them properly because currently it is very difficult to contribute to this cookbook code.

**Comment by zuazo** [comment]
Released in [`1.9.0`](https://github.com/zuazo/ssl_certificate-cookbook/releases/tag/1.9.0).

**Pull request merged**